### PR TITLE
replace validate_* with datatypes

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

**Comment by bastelfreak** [comment]
the goal of this is to get rid of the validate_* calls because they are deprecated. The goal is not to add datatypes to all parameters.

File: manifests/init.pp [hunk 1]
```diff
@@ -33,156 +33,129 @@
...
+  String $version                                                   = '6.4.1',
+  String $product                                                   = 'jira',
+  String $format                                                    = 'tar.gz',
+  Stdlib::Absolutepath $installdir                                  = '/opt/jira',
+  Stdlib::Absolutepath $homedir                                     = '/home/jira',
+  String $user                                                      = 'jira',
+  String $group                                                     = 'jira',
+  $uid                                                              = undef,
+  $gid                                                              = undef,
+  Stdlib::Absolutepath $shell                                       = '/bin/true',
```
Comment (alexjfisher) [inline manifests/init.pp]: If the aim is to improve readability, I'm not sure aligning *everything* is helpful.

Comment (alexjfisher) [inline manifests/init.pp]: Whilst googling, I came across this style too. https://github.com/jlambert121/jlambert121-puppet/blob/master/manifests/init.pp

Comment (bastelfreak) [inline manifests/init.pp]: @alexjfisher maybe we should merge this as is and discuss it on the mailing list?

**Pull request merged**